client/src/pages/Admin.js

```diff
@@ -0,0 +1,248 @@
+import axios from "axios";
+import PropTypes from "prop-types";
+import React, { Component } from "react";
+
+import AuthService from "../services/auth.service";
+
+const CSV_MIME = ["text/csv", "application/vnd.ms-excel"];
+const ZIP_MIME = ["application/zip", "application/x-zip-compressed"];
+
+class FileImporter extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      errors: [],
+      warnings: [],
+      selectedFile: null,
+      canConfirm: false,
+      imported: false,
+    };
+  }
+
+  /**
+   * Send the selected file
+   * @param {Event} e The event
+   */
+  sendSelectedFile(e) {
+    e.preventDefault();
+    const file = this.state.selectedFile;
+    const requestData = new FormData();
+
+    requestData.append("file", file);
+    if (this.state.canConfirm) {
+      requestData.append("confirmed", "true");
+    }
+
+    axios
+      .post(this.props.endpoint, requestData)
+      .then((res) => {
+        const warnings = res.data.warnings.map((w) => `[${w.code}] - ${w.message}`);
+        this.setState({
+          warnings,
+          errors: [],
+          canConfirm: !this.state.canConfirm,
+          imported: res.data.imported,
+        });
+      })
+      .catch((error) => {
+        if (error.response?.status === 400) {
+          const errors = error.response.data.errors.map((e) => `[${e.code}] - ${e.message}`);
+          this.setState({
+            errors,
+            warnings: [],
+            canConfirm: false,
+            imported: error.response.data.imported,
+          });
+          return;
+        }
+        this.setState({ errors: ["Une erreur est survenue de notre coté."] });
+      });
+  }
+
+  /**
+   * Check if the file format is valid
+   * @param {Event} e
+   */
+  handleFileChange(e) {
+    let file = e.target.files[0] || null;
+    const errors = [];
+
+    if (!this.props.mimeTypes.includes(file.type)) {
+      e.target.parentNode.reset();
+      file = null;
+      errors.push("Le format du fichier est invalide");
+    }
+    this.setState({ selectedFile: file, canConfirm: false, errors, imported: false });
+  }
+
+  /**
+   * Display a list
+   * @param {string[]} array The array to display
+   * @param {string} className The class of the list element
+   */
+  displayList(array, className) {
+    if (array.length === 0) {
+      return;
+    }
+    return (
+      <ul className={className}>
+        {array.map((element) => (
+          <li key={element}>{element}</li>
+        ))}
+      </ul>
+    );
+  }
+
+  render() {
+    return (
+      <div className="file-importer">
+        <form>
+          <input id="molecules-file" onChange={this.handleFileChange.bind(this)} type="file" />
+          <button
+            type="submit"
+            disabled={this.state.selectedFile === null}
+            onClick={this.sendSelectedFile.bind(this)}
```

Comment:
You're right about the submit event rather than the click event, but I think it's preferable to keep a name that makes it clear what the function does.